libs/community/langchain_community/retrievers/azure_ai_search.py

```diff
@@ -79,10 +86,17 @@ def _build_search_url(self, query: str) -> str:
 
     @property
     def _headers(self) -> Dict[str, str]:
-        return {
-            "Content-Type": "application/json",
-            "api-key": self.api_key,
-        }
+        if self.azure_ad_token_provider:
+            token = "Bearer " + f"{self.azure_ad_token_provider()}"
```

Comment:
```suggestion
            token = f"Bearer {self.azure_ad_token_provider()}"
```

The whole thing can be an f-string, no?